nixos/tests/matrix/matrix-alertmanager.nix

```diff
@@ -0,0 +1,150 @@
+              tls_private_key_path = "${key}";
+              enable_registration = true;
+              enable_registration_without_verification = true;
+              registration_shared_secret = "supersecret-registration";
+
+              listeners = [
+                {
+                  # The default but tls=false
+                  bind_addresses = [
+                    "0.0.0.0"
+                  ];
+                  port = 8448;
+                  resources = [
+                    {
+                      compress = true;
+                      names = [ "client" ];
+                    }
+                    {
+                      compress = false;
+                      names = [ "federation" ];
+                    }
+                  ];
+                  tls = false;
+                  type = "http";
+                  x_forwarded = false;
+                }
+              ];
+            };
+          };
+
+          networking.firewall.allowedTCPPorts = [ 8448 ];
+
+          environment.systemPackages = [
+            (pkgs.writeShellScriptBin "register_alertmanager_user" ''
+              exec ${pkgs.matrix-synapse}/bin/register_new_matrix_user \
+                -u alertmanager \
+                -p alertmanager-password \
+                --admin \
+                --shared-secret supersecret-registration \
+                http://localhost:8448
+            '')
+            # This is needed to solve a chicken and egg
+            # problem. Matrix-alertmanager expects a token for authentication,
+            # but a token is created after the user has been registered. This
+            # changes the token in the database to match the one specified in
+            # the service settings.
+            (pkgs.writers.writePython3Bin "hardcode_matrix_values"
+              {
+                libraries = with pkgs.python3Packages; [
+                  sqlite-utils
+                ];
+              }
+              ''
+                import sqlite3
+                con = sqlite3.connect("/var/lib/matrix-synapse/homeserver.db")
+                cur = con.cursor()
+                cur.execute(
+                    "update access_tokens set token='%s' where user_id = '%s'"
+                    % ("faketoken", "@alertmanager:homeserver")
+                )
+                con.commit()
+                con.close()
+              ''
+            )
+          ];
+        };
+
+      matrix_alertmanager =
+        { pkgs, ... }:
+        {
+          services.matrix-alertmanager = {
+            enable = true;
+            tokenFile = pkgs.writeText "token.txt" "faketoken";
+            secretFile = pkgs.writeText "secret.txt" "wontbeused";
+            homeserverUrl = "http://homeserver:8448";
+            matrixRooms = [
+              {
+                receivers = [ "matrix" ];
+                roomId = "!room_id:homeserver";
+              }
+            ]; # Not needed for matrix-alertmanager to start so use a fake one.
```

Comment:
This is not needed to start? Then why are we setting this at all?

Reply:
The comment should say "Doesn't need to exist for matrix-alertmanager to start". I rephrased it in a better way, thanks for catching it.

Reply:
The updated comment is great, thank you.